patches/server/0834-Fire-EntityChangeBlockEvent-in-more-places.patch

```diff
@@ -198,6 +198,35 @@ index 32995cb5efdad0bc34ecacacb78cccd21220ba8d..21212462e6b415e96536a27b2c009d15
                      level.setBlock(blockPos, blockState3, 11);
                      level.gameEvent(GameEvent.BLOCK_CHANGE, blockPos, GameEvent.Context.of(player, blockState3));
                      if (player != null) {
+diff --git a/src/main/java/net/minecraft/world/level/block/CakeBlock.java b/src/main/java/net/minecraft/world/level/block/CakeBlock.java
+index 49fe91a8eaeb2580c8ad0166e72540168af605f6..523e8982b8b981aafb4425b132e95d6f6c9eae4e 100644
+--- a/src/main/java/net/minecraft/world/level/block/CakeBlock.java
++++ b/src/main/java/net/minecraft/world/level/block/CakeBlock.java
+@@ -66,6 +66,11 @@ public class CakeBlock extends Block {
+                     itemstack.shrink(1);
+                 }
+ 
++                // Paper start - call change block event
++                if (!org.bukkit.craftbukkit.event.CraftEventFactory.callEntityChangeBlockEvent(player, pos, CandleCakeBlock.byCandle(block))) {
++                    return InteractionResult.PASS;
++                }
++                // Paper end - call change block event
+                 world.playSound((Player) null, pos, SoundEvents.CAKE_ADD_CANDLE, SoundSource.BLOCKS, 1.0F, 1.0F);
+                 world.setBlockAndUpdate(pos, CandleCakeBlock.byCandle(block));
+                 world.gameEvent((Entity) player, GameEvent.BLOCK_CHANGE, pos);
+@@ -107,6 +112,12 @@ public class CakeBlock extends Block {
+             int i = (Integer) state.getValue(CakeBlock.BITES);
+ 
+             world.gameEvent((Entity) player, GameEvent.EAT, pos);
++            // Paper start - call change block event
++            final BlockState newState = i < 6 ? state.setValue(CakeBlock.BITES, i + 1) : world.getFluidState(pos).createLegacyBlock();
```

Comment:
Do we need the fluid legacy check even if the block isn't waterloggable?

Reply:
We do that in other cases so it's consistent since that is what removeBlock/destroyBlock does.

Reply:
👍